#!/usr/bin/env python
PACKAGE = "morus_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("vz_kp", double_t, 0, "VZ ctl PID P gain", 40, 0, 100)
gen.add("vz_ki", double_t, 0, "VZ ctl PID I gain", 0.1, 0, 100)
gen.add("vz_kd", double_t, 0, "VZ ctl PID D gain", 0.0, 0, 100)

gen.add("z_kp", double_t, 0, "Z ctl PID P gain", 4, 0, 100)
gen.add("z_ki", double_t, 0, "Z ctl PID I gain", 0.0, 0, 100)
gen.add("z_kd", double_t, 0, "Z ctl PID D gain", 1.0, 0, 100)

gen.add("z_comp_b0", double_t, 0, "Z compensator b0", 0.3, -10, 10)
gen.add("z_comp_b1", double_t, 0, "Z compensator b1", -0.299, -10, 10)
gen.add("z_comp_a1", double_t, 0, "Z compensator a1", 1, -10, 10)
gen.add("z_comp_gain", double_t, 0, "Z compensator gain", 0.05, 0, 100)

gen.add("vz_comp_b0", double_t, 0, "VZ compensator b0", 1, -10, 10)
gen.add("vz_comp_b1", double_t, 0, "VZ compensator b1", -0.998, -10, 10)
gen.add("vz_comp_a1", double_t, 0, "VZ compensator a1", 1, -10, 10)
gen.add("vz_comp_gain", double_t, 0, "VZ compensator gain", 100, 0, 300)

gen.add("switch_eps", double_t, 0, "Switching constant", 0.01, 0, 1)
gen.add("z_beta", double_t, 0, "Z switch gain", 0.01, 0, 1)
gen.add("vz_beta", double_t, 0, "VZ switch gain", 30, 0, 200)

gen.add("z_ff", double_t, 0, "Z feed-forward gain", 1, 0, 10)
gen.add("vz_ff", double_t, 0, "VZ feed-forward gain", 8, 0, 50)

exit(gen.generate(PACKAGE, "morus_control", "SmcMmcuavPositionCtlParams"))